File: Java/src/main/java/com/gildedrose/handler/ConjuredItemHandler.java
package com.gildedrose.handler;

import com.gildedrose.Item;
import com.gildedrose.utility.ItemUtilities;

public class ConjuredItemHandler implements ItemHandler {
    @Override
    public boolean handles(String itemName) {
        return itemName != null && itemName.toLowerCase().contains("conjured");
    }

    @Override
    public void updateItem(Item item) {
        for (int i = 0; i < 2; i++) {
            if (ItemUtilities.canDecreaseQuality(item)) {
                ItemUtilities.decreaseQuality(item);
            }
        }

        ItemUtilities.decreaseSellIn(item);

        if (ItemUtilities.isExpired(item)) {
            for (int i = 0; i < 2; i++) {
                if (ItemUtilities.canDecreaseQuality(item)) {
                    ItemUtilities.decreaseQuality(item);
                }
            }
        }
    }
}

File: TypeScript/app/handler/ConjuredItemHandler.ts
import { Item } from '../gilded-rose';
import { ItemHandler } from './ItemHandler';
import { ItemUtils } from '../utility/ItemUtils';

export class ConjuredItemHandler implements ItemHandler {
  handles(item: Item): boolean {
    return item.name.toLowerCase().includes('conjured');
  }

  updateItemQuality(item: Item): void {
    if (item.quality > 0) {
      ItemUtils.decreaseQuality(item, 2);
    }

    ItemUtils.decreaseSellIn(item);

    if (item.sellIn < 0 && item.quality > 0) {
      ItemUtils.decreaseQuality(item, 2);
    }
  }
}
